extensions/ql-vscode/test/pure-tests/remote-queries/gh-api/gh-api-client.test.ts

```diff
@@ -0,0 +1,93 @@
+import { expect } from 'chai';
+import * as path from 'path';
+
+import * as Octokit from '@octokit/rest';
+import { retry } from '@octokit/plugin-retry';
+
+import { setupServer } from 'msw/node';
+
+import {
+  getRepositoryFromNwo,
+  getVariantAnalysis,
+  getVariantAnalysisRepo, getVariantAnalysisRepoResult,
+  submitVariantAnalysis
+} from '../../../../src/remote-queries/gh-api/gh-api-client';
+import { Credentials } from '../../../../src/authentication';
+import {
+  createMockSubmission
+} from '../../../../src/vscode-tests/factories/remote-queries/shared/variant-analysis-submission';
+import { createRequestHandlers } from '../../../../src/mocks/request-handlers';
+
+const mockCredentials = {
+  getOctokit: () => Promise.resolve(new Octokit.Octokit({ retry }))
+} as unknown as Credentials;
+
+const server = setupServer();
+
+before(() => server.listen());
+
+afterEach(() => server.resetHandlers());
+
+after(() => server.close());
+
+async function loadScenario(scenarioName: string) {
+  const handlers = await createRequestHandlers(path.join(__dirname, '../../../../src/mocks/scenarios', scenarioName));
+
+  server.use(...handlers);
+}
+
+describe('submitVariantAnalysis', () => {
+  it('returns the submitted variant analysis', async () => {
+    await loadScenario('problem-query-success');
+
+    const result = await submitVariantAnalysis(mockCredentials, createMockSubmission());
+
+    expect(result).not.to.be.undefined;
+    expect(result.id).to.eq(146);
+  });
+});
+
+describe('getVariantAnalysis', () => {
+  it('returns the variant analysis', async () => {
+    await loadScenario('problem-query-success');
+
+    const result = await getVariantAnalysis(mockCredentials, 557804416, 146);
```

Comment:
What do you think about getting these ids from the scenario data instead of hard-coding in a test as well? I'm not sure if that would be super ugly so happy to be told no :)

Reply:
I've implemented it and I don't think it's ugly, but I did have to change how the `tsconfig.json` for tests was loaded.